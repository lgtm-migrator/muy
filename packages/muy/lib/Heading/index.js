import React from "react"
import Typography from "@material-ui/core/Typography"

export const defaultProps = {
  level: 1,
}

const Heading = ({ level = defaultProps.level, ...props }) => (
  <Typography component={`h${level}`} variant={`h${level}`} {...props} />
)

export default Heading

Heading.defaultProps = defaultProps

export const H1 = props => <Heading level={1} {...props} />
export const H2 = props => <Heading level={2} {...props} />
export const H3 = props => <Heading level={3} {...props} />
export const H4 = props => <Heading level={4} {...props} />
export const H5 = props => <Heading level={5} {...props} />
export const H6 = props => <Heading level={6} {...props} />
