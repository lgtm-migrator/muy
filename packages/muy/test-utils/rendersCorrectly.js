import renderer from "react-test-renderer"

const rendersCorrectly = component => {
  const tree = renderer.create(component).toJSON()
  return expect(tree).toMatchSnapshot()
}

export default rendersCorrectly
