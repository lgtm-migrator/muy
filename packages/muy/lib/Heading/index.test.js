import React from "react"
import Heading, { defaultProps } from "."
import rendersCorrectly from "../../test-utils/rendersCorrectly"

describe("Heading", () => {
  describe("default props", () => {
    it("should be 1 for level", () => {
      expect(defaultProps.level).toEqual(1)
    })
  })
  it("should render correctly with no properties", () => {
    rendersCorrectly(<Heading />)
  })
  it("should render correctly with level 1 and no props", () => {
    rendersCorrectly(<Heading level={1} />)
  })
  it("should render correctly with level 1 and children", () => {
    rendersCorrectly(<Heading level={1}>Heading text</Heading>)
  })
})
