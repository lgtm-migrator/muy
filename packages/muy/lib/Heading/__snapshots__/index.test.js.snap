// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading should render correctly with level 1 and children 1`] = `
<h1
  className="MuiTypography-root MuiTypography-h1"
>
  Heading text
</h1>
`;

exports[`Heading should render correctly with level 1 and no props 1`] = `
<h1
  className="MuiTypography-root MuiTypography-h1"
/>
`;

exports[`Heading should render correctly with no properties 1`] = `
<h1
  className="MuiTypography-root MuiTypography-h1"
/>
`;
